samples/isprime/isprime.neon

```diff
@@ -0,0 +1,211 @@
+    VAR eval: Number := 0
+    VAR nums: String := ""
+    VAR vals: Array<String> := []
+
+    FOREACH p IN sys.args INDEX i DO
+        IF p = "-q" OR p = "-Q" THEN
+            bQuiet := TRUE
+        ELSIF p = "-k" OR p = "-K" THEN
+            TestIterations := num(sys.args[i+1])
+        ELSE
+            nums := sys.args[i]
+        END IF
+    END FOREACH
+
+    IF TestIterations # iDefaultTests THEN
+        % Wait until after all the command line params are parsed, just in case we are supposed to be quiet.
+        output("Using \(TestIterations) iteration tests.")
+    END IF
+
+    VAR hasDash: Number := string.find(nums, "-")
+    vals := string.split(nums, "-")
+    IF string.hasPrefix(nums, "-") THEN
+        % Print the last prime number BEFORE n.  -number
+        eval := 1
+        sval := num(vals[0])
+        output("Locating last prime number before \(sval)...")
+        VAR val: Number := 0
+        FOR x := sval TO eval STEP -1 DO
+            val := x
+            IF RabinMiller(val, TestIterations) THEN
+                IF val = num(vals[0]) THEN
+                    NEXT FOR
+                END IF
+                output("\(val) is the last prime before \(vals[0]).")
+                IF bQuiet THEN
+                    print("\(val)")
+                END IF
+                sys.exit(0)
+            END IF
+        END FOR
+        output("There is no prime number before \(sval).")
+    ELSIF string.hasSuffix(nums, "-") THEN
+        % Print the next prime AFTER n.   number-
+        IF hasDash > 0 THEN
+            sval := num(substring(vals[0], 0, hasDash))
+            output("Locating the next prime number after \(sval)...")
+            WHILE TRUE DO
+                IF isOdd(sval) THEN
+                    sval := sval + 2
+                ELSE
+                    INC sval
+                END IF
+                IF RabinMiller(sval, TestIterations) THEN
+                    EXIT WHILE
+                END IF
+            END WHILE
+            output("\(sval) is the next prime.")
+            IF bQuiet THEN
+                print("\(sval)")
+            END IF
+        END IF
+    ELSIF hasDash > 0 THEN
+        % Print pimes within a RANGE...  number-number
+        IF hasDash > 0 THEN
+            sval := num(vals[0])
+            eval := num(vals[1])
+            output("Locating all prime numbers between \(sval) and \(eval)...")
+            VAR val: Number := 0
+            FOR x := sval TO eval DO
+                val := x
+                IF RabinMiller(val, TestIterations) THEN
+                    print("\(val)")
+                END IF
+                IF isOdd(val) THEN
+                    val := val + 2
+                ELSE
+                    INC val
+                END IF
+            END FOR
+        END IF
+    ELSE
+        sval := num(nums)
+        output("Testing \(sval) for primality.")
+        IF RabinMiller(sval, TestIterations) THEN
+            print("\(sval) is prime.")
```

Comment:
No, only because this is a different test, this directly tests a number for prime, while the others search for a prime.  This can be argued, however, that in quiet mode, it JUST output the number, but not quiet mode it outputs what I have here.  I was torn by this when I wrote it.